service.py

```diff
@@ -436,10 +463,12 @@ def post_article():
     if request.method == 'POST':
         email = request.form['email']
         # means that author isn't in database now
-        if not Author.query.filter_by(mail=email).first():
-            author = Author(mail=email, is_banned=False)
-            db.session.add(author)
-
+        if sendMailAndVerify(email):
```

Comment:
@pan0303 This checks email validity before adding the author.  Why don't you also check email validity before one posts an article?  -Hui